ChalkLab/SciDataLib
# Adds GitHub Actions integration to setup CI + formatting fixes from linting

Closes #18 

Work includes:
 - Adding a GitHub Actions yaml file with the correct directory so GitHub Actions picks it up automatically and will kick off CI based on instuctions in this yaml file (`.github/workflows/actions.yml`). See [here](https://docs.github.com/en/actions/quickstart) for quickstart guide.
 - In the CI, setting up different python versions to test, installing poetry, installing the project requirements, and then running linting via flake8
 - Format fixes to the project based on the output of linting (for the issues it found, see [this build](https://github.com/ChalkLab/SciDataLib/runs/1913708740?check_suite_focus=true))
 - Moved the `example.py` script to the top-level since it is not "part of the package" and instead is a script that "uses" the library.

To test:
 - I ran the `example.py` via `poetry run python example.py` just to check it worked but may want to validate the output since I modified the file via linting. We probably want to use this file to write actual tests as part of issue #19 in the future. That way we turn it from a manual tests (we run it and check visually) to an automated test (we can run via `pytest` and also let CI run it once I add the test stage for every check-in of code)
 - Navigate to the "Actions" tab at the top of the project page and explore what is going on. I am more than happy to answer any questions and can hop on a video call if you would like to discuss w/ screen sharing.
![image](https://user-images.githubusercontent.com/17032543/108123399-cbf03880-7073-11eb-8a4e-110e12f42b1b.png)
